examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.cpp

```diff
@@ -632,115 +627,82 @@ static void wfx_rsi_save_ap_info() // translation
  **********************************************************************************************/
 static sl_status_t wfx_rsi_do_join(void)
 {
-    sl_status_t status = SL_STATUS_OK;
-    sl_wifi_security_t connect_security_mode;
+    ReturnErrorCodeIf((wfx_rsi.dev_state & (WFX_RSI_ST_STA_CONNECTING | WFX_RSI_ST_STA_CONNECTED)), SL_STATUS_IN_PROGRESS);
+    sl_status_t status                = SL_STATUS_OK;
+    sl_wifi_client_configuration_t ap = { 0 };
```

Comment:
In C++, this doesn't do the same thing as in C99. This only sets the first element of the struct to 0. It doesn't guarantee full zero initialization of the entire struct and null termination of the strings.

Since these are plain-old-data structs, you can memset to zero (assuming zero is valid for everything). I noticed that the sl_wifi_types.h has no initializers inline of the structures, so I guess it's C-compatible code, which means it's important to be clean with your initializations.

And this is why it's dangerous to have nul-termination expectations in complex data structures...

Reply:
using `memset()`